internal/e2e/e2eutil/compare_documents.go

```diff
@@ -99,18 +107,98 @@ func getFile(script *testscript.TestScript, filePath string) *os.File {
 	return file
 }
 
-func compareMetaData(script *testscript.TestScript, first, second *sbom.Metadata) bool {
-	firstStr := first.String()
-	secondStr := second.String()
+func compareMetaData(script *testscript.TestScript, have, want *sbom.Metadata) bool {
```

Comment:
Could this not just be something like

```go
haveBytes, err := proto.MarshalOptions{Deterministic: true}.Marshal(have)
if err != nil {
    // handle error
}

wantBytes, err := proto.MarshalOptions{Deterministic: true}.Marshal(want)
if err != nil {
    // handle error
}

return haveBytes == wantBytes
```

Reply:
Quick test didn't work, will dig a bit more. Would this handle the timestamp? Thats the biggest reason for doing the individual field checks

Reply:
I think for the time being we'll have to keep the individual checks, the source data hash and the timestamp are going to always be different

Reply:
In the past for situations like this, I've removed the timestamp from both before comparison. It doesn't look like you're comparing timestamps between the two anyway, so it's effectively the same thing

Reply:
Oh cool, didn't think of doing that. I'll make that change

Reply:
Ok so there are 3 fields that are expected to be different. (apologies for the back and forth, it's been a bit since I looked at this originally)

1. MetaData.SourceData.Hash (and presumably size)
2. MetaData.Date
3. MetaData.Tools (protobom appends an entry here)

Following your example, I was able to get it working, with a couple additional steps to account for all 3 fields listed above. I'll push that change momentarily. It should be noted that the addition of a protobom tool will not be needed once the without modification changes take place so i wonder if it's better to nil out that field as well. Thoughts?

Reply:
LGTM!

Reply:
Also as an side comment, the addition of `protobom-devel` Tool only happens in spdx documents, as my code suggests, but I'm not sure why that is. maybe worth an issue in protobom?

And I think it's supposed to be `protobom-v0.5.0` (or match the go mod in general) and I've only ever seen `protobom-devel`. Could be an easy bug to fix also